### ISSUE TITLE
Exception when computing Alternative

### ISSUE THREAD

**Issue by codymccoy** [HydrologicEngineeringCenter/HEC-FDA#712]
I am wondering if this has to do with the update to how the histograms read and write the bin counts to xml.
The issue is that the bin counts is 0.
![image](https://user-images.githubusercontent.com/47757848/208982858-afb95316-0922-4653-91fd-87e94072cd60.png)
![image](https://user-images.githubusercontent.com/47757848/208982871-cb4c1259-261d-4749-8942-3fe4756c06ef.png)

I am using the Beargrass Creek study.
![image](https://user-images.githubusercontent.com/47757848/208983134-3d5124e6-7879-4ed6-8f56-10de8c44ab66.png)

**Comment by codymccoy** [comment]
@rnugent3 Is it possible to have bin counts of zero? I'm thinking that maybe this is an issue when using studies that where created before the recent xml change.

**Comment by rnugent3** [comment]
bin counts of zero shouldn't be a problem.

**Comment by rnugent3** [comment]
what is the index in the image above?

**Comment by rnugent3** [comment]
@codymccoy would you send me the study?

**Comment by codymccoy** [comment]
@rnugent3 I think you were the one that sent it to me:
![image](https://user-images.githubusercontent.com/47757848/209003886-75499e4c-9390-4c64-b19c-c5708daee0fd.png)

**Comment by codymccoy** [comment]
[HEC-FDA_Version-2_Study.zip](https://github.com/HydrologicEngineeringCenter/HEC-FDA/files/10281577/HEC-FDA_Version-2_Study.zip)

**Comment by codymccoy** [comment]
I tried re-computing the "without" scenarios which was successful and then i tried computing the alternative and ran into a divide by zero:
![image](https://user-images.githubusercontent.com/47757848/209004411-543ea020-97bd-4aa1-a6ee-fc0bfd9cb1ba.png)

**Comment by rnugent3** [comment]
oh boy. I am looking into it.

**Comment by rnugent3** [comment]
I am successfully computing scenarios and alternatives on main

**Comment by codymccoy** [comment]
ok, interesting. I bet it has something to do with the xml code that I changed then.

**Comment by codymccoy** [comment]
I think i have fixed this issue. I had changed the way the bin counts write and read to xml and didn't support the old way that it was reading from xml. I added an "else" statement that uses the old way. I made this change to both histogram and threadsafeInline...

![image](https://user-images.githubusercontent.com/47757848/209215622-b5ac77d4-376e-4463-b24d-99e764f37b97.png)

This is fixed on branch Update_Index
